src/twisted/internet/endpoints.py

```diff
@@ -694,26 +717,43 @@ def connect(self, protocolFactory):
         connection which is established first.
         """
         wf = protocolFactory
-        d = self._nameResolution(self._host, self._port)
+        d = Deferred()
+        if self._badHostname:
+            ude = ValueError("invalid hostname: {}".format(self._hostText))
+            d.errback(ude)
+            return d
+        addresses = []
+        @provider(IResolutionReceiver)
+        class MyReceiver(object):
+            @staticmethod
+            def resolutionBegan(resolutionInProgress):
+                pass
+            @staticmethod
+            def addressResolved(address):
+                addresses.append(address)
+            @staticmethod
+            def resolutionComplete():
+                d.callback(addresses)
```

Comment:
This is correct behavior on TwistedChecker's part.  3 lines between top-level suites, 2 lines between methods on top-level classes.  There are no required newlines between nested callbacks - and inner classes are just a kind of nested callback.  The VWS requirements are to make it easy to scan a file and see where the globally-visible methods are; within a function newlines would be confusing because they'd imply top-level stuff was being defined.